src/main/java/ua/team3/carsharingservice/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package ua.team3.carsharingservice.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import ua.team3.carsharingservice.dto.CarDto;
+import ua.team3.carsharingservice.dto.CreateCarRequestDto;
+import ua.team3.carsharingservice.exception.EntityNotFoundException;
+import ua.team3.carsharingservice.mapper.CarMapper;
+import ua.team3.carsharingservice.model.Car;
+import ua.team3.carsharingservice.repository.CarRepository;
+import ua.team3.carsharingservice.service.CarService;
+
+@Service
+@RequiredArgsConstructor
+public class CarServiceImpl implements CarService {
+    private final CarRepository carRepository;
+    private final CarMapper carMapper;
+
+    @Override
+    public CarDto addCar(CreateCarRequestDto requestDto) {
+        Car car = carMapper.toModel(requestDto);
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    public List<CarDto> findAllCars(Pageable pageable) {
+        return carRepository.findAll(pageable).stream()
+                .map(carMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CarDto findCarById(Long id) {
+        return carRepository.findById(id)
+                .map(carMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find car with id: " + id));
+    }
+
+    @Override
+    public CarDto updateById(Long id, CreateCarRequestDto requestDto) {
+        Car car = carRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find car with id: " + id));
+        car.setBrand(requestDto.getBrand());
+        car.setType(Car.CarType.valueOf(requestDto.getType()));
+        car.setInventory(requestDto.getInventory());
+        car.setDailyFee(requestDto.getDailyFee());
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        carRepository.findById(id).ifPresentOrElse(carRepository::delete, () -> {
+            throw new EntityNotFoundException("Can't find car with id: " + id);
+        });
+    }
```

Comment:
I think it should be separated into few lines. Somehow like here (approximately):
```suggestion
        carRepository.findById(id)
                .ifPresentOrElse(carRepository::delete, 
                    () -> {
                        throw new EntityNotFoundException("Can't find car with id: " + id);
                }
        );
    }
```

Reply:
also, add this
```suggestion
        carRepository.findById(id).ifPresentOrElse(carRepository::delete, () -> {
            throw new EntityNotFoundException("Can't find car with id: " + id, e);
        });
    }
```

Reply:
But what is `e` here?

Reply:
We were talking about it. The ifPresentOrElse method of Optional does not accept any parameter in Runnable, which is used if no value is found. Therefore, it is impossible to pass an additional argument (for example, e) to the lambda expression.